### PR TITLE
refactor: move WCAG metadata from test name to class name

### PR DESCRIPTION
**Pull request by mohanraj-r**
### Automatic checks 🤖 
* Add ability and convenience shortcut for excluding files from automatic checks
* Rename run only files option to files filter to better reflect the functionality 
  * `AutoCheckOpts.runOnlyOnFiles` -> `AutoCheckOpts.filesFilter`
  * `process.env.SA11Y_AUTO_ONLY` ->  `process.env.SA11Y_AUTO_FILTER`

### Results processor
* Move the WCAG metadata and Priority info to class name in Jest results so that downstream CI systems make use of the info accordingly
  - Remove WCAG version, Level for brevity
  - Include only WCAG SC and Priority as they are the more essential info
* Error msgs: Move help URL up in order to prevent it from getting truncated e.g. when CSS selectors are long

## Fix 🐛 
* WDIO: Fix race condition when loading axe using poll with timeout

### CODE REVIEW

File: packages/jest/src/resultsProcessor.ts [hunk 1]
```diff
@@ -57,10 +57,11 @@ function processA11yErrors(testSuite: TestResult, testResult: AssertionResult) {
             // TODO (spike) : What happens if there are ever multiple failureDetails?
             //  Ideally there shouldn't be as test execution should be stopped on failure
             A11yResults.add(error.a11yResults, suiteName).forEach((a11yResult) => {
+                const className = `${suiteName}:[${a11yResult.wcag}]`;
```
Comment (trevor-bliss) [inline packages/jest/src/resultsProcessor.ts:60]: Is this the separator referenced from the `jest-tool` PR? If so, let's pull it out as a constant so it's more clear it has significance.

Comment (mohanraj-r) [reply packages/jest/src/resultsProcessor.ts]: Yes, just made those changes - pushing them now. Initially thought of exporting the const and importing it in jest-tool, but don't think the adding the import dependency overhead to jest-tool is justified in this case.

File: packages/wdio/src/wdio.ts [hunk 2]
```diff
@@ -33,16 +33,17 @@ export async function getAxeVersion(driver: WdioBrowser): Promise<typeof axeVers
 }
 
 /**
- * Load axe source into browser if it is not already loaded and return version of axe
+ * Load axe source into browser if it is not already loaded and return version of axe.
+ * Keep waiting and trying until given timeout milliseconds.
  */
-export async function loadAxe(driver: WdioBrowser): Promise<void> {
-    if ((await getAxeVersion(driver)) !== axeVersion) {
+export async function loadAxe(driver: WdioBrowser, timeout = 1000, pollTime = 100): Promise<void> {
```
Comment (trevor-bliss) [inline packages/wdio/src/wdio.ts:39]: Is one second a long enough default timeout? This function executes Javascript in the browser. Feels like that time could vary greatly depending on the target browser.

**Review by trevor-bliss — APPROVED** [review]
Consider increasing the default timeout, but up to you depending on average times you currently see.

File: packages/jest/README.md [hunk 3]
```diff
@@ -154,10 +154,14 @@ Automatic checks can also be enabled using environment variables
...
--   Invoking `jest` with environment variables as above will enable automatic checks with no changes required to `setup()`
--   The environment variables can be used to set up parallel builds e.g., in a CI environment without code changes to `setup()` to opt-in to automatic checks
--   Setting `SA11Y_DEBUG=1` will output verbose logging
--   `SA11Y_AUTO_ONLY` can be used to specify a comma seperated list of files to run automatic checks on while other files are ignored.
+- Invoking `jest` with environment variables as above will enable automatic checks with no changes required to `setup()`
+- The environment variables can be used to set up parallel builds e.g., in a CI environment without code changes to `setup()` to opt-in to automatic checks
+- Setting `SA11Y_DEBUG=1` will output verbose logging
+- `SA11Y_AUTO_FILTER` can be used to specify a comma seperated list of test file paths to filter for automatic checks
+  - When specified, automatic checks will be run only on given files
+  - Files can be excluded from automatic checks by prefixing the filename with `!`
```
Comment (cordeliadillon) [inline packages/jest/README.md:162]: Do you need a ! in front of each excluded test, or just one in front of the comma-separated list?

**Review by cordeliadillon — APPROVED** [review]

File: packages/jest/README.md (hunk 3 above)
Comment (mohanraj-r) [reply packages/jest/README.md]: Good point, changed doc to make it explicit that ! is needed for each file

File: packages/wdio/src/wdio.ts (hunk 2 above)
Comment (mohanraj-r) [reply packages/wdio/src/wdio.ts]: Agree, increased the timeout and changed the polling logic to load axe only once.

**Review by mohanraj-r — COMMENTED** [review]
@trevor-bliss @cordeliadillon 
Thanks for the review. Have taken care of the comments. Please review the updates.

**Review by trevor-bliss — APPROVED** [review]

**Review by cordeliadillon — APPROVED** [review]

**Pull request merged**